File: backend/src/main/java/com/yigongil/backend/domain/study/Role.java
package com.yigongil.backend.domain.study;

public enum Role {

    MASTER(0),
    STUDY_MEMBER(1),
    APPLICANT(2),
    NO_ROLE(3),
    ;

    private final int code;

    Role(int code) {
        this.code = code;
    }

    public int getCode() {
        return code;
    }
}

File: android/domain/src/main/java/com/created/domain/model/Role.kt
package com.created.domain.model

enum class Role {
    MASTER, STUDY_MEMBER, APPLICANT, NOTHING, GUEST;

    companion object {
        fun valueOf(index: Int): Role {
            return Role.values().find { role: Role -> role.ordinal == index }
                ?: throw IllegalArgumentException("해당 역할을 찾을 수 없습니다.")
        }
    }
}
